File: src/FsApi/AppSettings.fs
namespace Api

[<CLIMutable>]
type AuthSettings = {
    Issuer: string
    Secret: string
}

[<RequireQualifiedAccess>]
type StoreType =
    | Local = 0

[<CLIMutable>]
type StoreSettings = {
    Type: StoreType
    Uri: string
}

[<CLIMutable>]
type AppSettings = {
    Auth: AuthSettings
    Store: StoreSettings
}

File: src/CsApi/AppSettings.cs
namespace Api;


public record AuthSettings {
    public required string Issuer { get; init; }
    public required string Secret { get; init; }
}

public enum StoreType {
    Local = 0
}

public record StoreSettings {
    public required StoreType Type { get; init; }
    public required string Uri { get; init; }
}


public record AppSettings {
    public required AuthSettings Auth { get; init; }
    public required StoreSettings Store { get; init; }
}
